app/models/assessment.rb

```diff
@@ -13,33 +13,35 @@ class Assessment < ApplicationRecord
   has_one :reported_condition, class_name: 'ReportedCondition'
   belongs_to :patient
 
-  scope :twenty_four_hours_fever_or_fever_medication, lambda {
-    where('created_at >= ?', 24.hours.ago).where_assoc_exists(:reported_condition, &:fever_or_fever_medication)
-  }
+  after_save :update_patient_linelist_after_save
+  before_destroy :update_patient_linelist_before_destroy
 
-  scope :seventy_two_hours_fever_or_fever_medication, lambda {
-    where('created_at >= ?', 72.hours.ago).where_assoc_exists(:reported_condition, &:fever_or_fever_medication)
-  }
-
-  scope :ten_days_symptomatic, lambda {
-    where('created_at >= ?', 10.days.ago).where(symptomatic: true)
-  }
-
-  scope :older_than_seventy_two_hours, lambda {
-    where('created_at <= ?', 72.hours.ago)
-  }
+  def update_patient_linelist_after_save
+    patient.update(
+      symptom_onset: patient.assessments.where(symptomatic: true).minimum(:created_at),
```

Comment:
same thing in that it needs to work on update as well. there's actually a bug in our current implementation of the way we update symptom onset because it doesn't update when assessments are updated from symptomatic to asymp.

for example, let's say a monitoree completed a symptomatic assessment and then an epi marked it as reviewed or edited it to be asymptomatic. the symptom onset date should be reset to nil because the symptom onset date is defined as the date of the first symptomatic assessment (which there are no longer any of).

Reply:
It makes sense to have this `after_save` but I'm just saying that isn't the symptom onset `this.created_at if this.symptomatic && this.patient.symptom_onset.nil?`

Reply:
that doesn't handle the case of assessment updates though